python/tvm/micro/contrib/stm32/emitter.py

```diff
@@ -0,0 +1,1557 @@
+    except KeyError:
+        raise ValueError(f"Data type {dltype} is not supported")
+
+
+def _get_aligned_offset(offset, dltype):
+    align = _get_type_size(dltype)
+    if offset % align != 0:
+        offset = offset + (align - offset % align)
+    return offset
+
+
+def _get_num_tensor_elts(shape):
+    size = 1
+    for dim in shape:
+        size = size * dim
+    return size
+
+
+def _get_tensor_size_bytes(dims, dltype):
+    size = _get_num_tensor_elts(dims)
+    return size * _get_type_size(dltype)
+
+
+def _preprocess_code(src):
+    """ Hack the C code implementing the model. """
+
+    # TODO: '#include "stm32lib.h"\n\n' - the 'magic option is not contributed yet
+    dst = "#include <stdio.h>\n" "#include <math.h>\n\n"
+    for line in src.splitlines():
+        #
+        # This is sort of hacking - when AoT is available, we will be
+        # able to clean this ...
+        #
+        dst = dst + line + "\n"
+
+    return dst
+
+
+class CodeEmitter(object):
+    """Code emitter class."""
+
+    #
+    # Constants:
+    #
+    DATA_ALIGNMENT_BYTES = 8
+
+    def __init__(self, include_activations=True, include_inputs=True, include_outputs=True):
+        """Initialize the Emitter instance.
+
+        Parameters
+        ----------
+        include_activations:
+            The Emitter allocates the storage for the activations data
+            and places it in a specific data section. If Falsr, the
+            main application is responsible for allocating the activations
+            storage. Default: True.
+
+        include_inputs/include_outputs:
+            The Emitter allocates the storage for the input/output data.
+            This storage is shared with the activations and placed in the
+            specific activations data section. If False, the main
+            application is responsible for allocating the input/output
+            data storage. Default: True.
+
+        Returns
+        -------
+            CodeEmitter object.
+
+        """
+
+        #
+        # Static model: activations placed into a nn_data_act section
+        # Dynamic model: activations need to be malloc'ed by the
+        #   applications.
+        #
+        self.activations_static = include_activations
+
+        #
+        # Inputs/outputs may be allocated within the activations or
+        # separately.
+        # TODO: Separate the inputs from activations inside TVM.
+        #
+        if include_inputs is True:
+            assert (
+                self.activations_static == True
+            ), "###Error: Static inputs are not allowed without activations."
+        self.inputs_static = include_inputs
+
+        if include_outputs is True:
+            assert (
+                self.activations_static == True
+            ), "###Error: Static outputs are not allowed without activations."
+        self.outputs_static = include_outputs
+
+        #
+        # Parsed graph
+        #
+        self.nodes_ = []
```

Comment:
this is a bit of a nit, but in general elsewhere in `python/` we prepend the `_` for private members e.g. `self._nodes`. do you mind updating this?